conans/client/graph/conanfile_dependencies.py

```diff
@@ -57,7 +58,7 @@ def host_requires(self):
                     if not edge.build_require and not edge.private]
         requires.extend([ConanFileInterface(edge.dst.conanfile)
                          for edge in self._node.dependencies if edge.build_require])
-        return DependencyOrderedSet(requires)
+        return DependencyOrderedSet([r for r in requires if r.context == CONTEXT_HOST])
```

Comment:
@memsharded BIG check on this. I'm not sure what I'm doing :sweat_smile:

Reply:
Yes, this seems correct, and a bug, I will do a separate PR for this.

Reply:
https://github.com/conan-io/conan/pull/8823